atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java

```diff
@@ -627,26 +629,41 @@ private boolean replaceExistingEntry(
    */
   private boolean appendEntry(
       final long index,
-      final PersistedRaftRecord entry,
+      final ReplicatableRecord entry,
       final CompletableFuture<AppendResponse> future) {
     try {
       final IndexedRaftLogEntry indexed;
-      indexed = raft.getLog().append(entry);
+      if (entry instanceof PersistedRaftRecord raftRecord) {
+        indexed = raft.getLog().append(raftRecord);
+      } else if (entry instanceof ReplicatableJournalRecord serializedJournalRecord) {
+        indexed = raft.getLog().append(serializedJournalRecord);
+      } else {
+        throw new IllegalStateException(
```

Comment:
:thought_balloon: This is where sealed types will become useful :) Right now with `if` the compiler doesn't quite get it, unfortunately

Reply:
Unfortunately, I had to remove the sealed because it was breaking Kryo serializaiton. RollingUpdateTest was failing. So I had to revert it. https://github.com/camunda/zeebe/pull/13143/commits/a0f1a463f8026788e31049b44aee0f343f9ca72e

Reply:
:exploding_head: Sealed breaks Kryo?

Reply:
What if we made the class a record? Then it's fine no, as it's by definition final, and Kryo can handle it? But making it a record is a breaking change according to Kryo, so it would have to be done before the release. Have we released yet with `PersistedRaftRecord`?